test/cli.test.ts

```diff
@@ -0,0 +1,112 @@
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { prepareEnvironment } from '@gmrchk/cli-testing-library'
+import { expect } from 'chai'
+
+describe('CLI Testing', () => {
+  it('Detect config file changes successfully', async () => {
+    const { spawn, cleanup, writeFile } = await prepareEnvironment()
+
+    const initFile = `
+probes:
+  - id: 1
+    name: Google
+    interval: 1
+    incidentThreshold: 6
+    recoveryThreshold: 6
+    requests:
+      - url: https://google.com
+notifications:
+  - id: unique-id-monika-notif,
+    type: desktop
+`
+    const changeFile = `
+probes:
+  - id: 1
+    name: Google
+    interval: 2
+    incidentThreshold: 4
+    recoveryThreshold: 4
+    requests:
+      - url: https://google.com
+  - id: 2
+    name: Github
+    interval: 2
+    incidentThreshold: 4
+    recoveryThreshold: 4
+    requests:
+      - url: https://github.com
+notifications:
+- id: unique-id-monika-notif,
+  type: desktop
+`
+
+    // write initial yml
+    await writeFile('./cli-test.yml', initFile)
+
+    // run monika
+    const { getStdout, wait, kill } = await spawn(
+      'node',
+      './bin/run -c cli-test.yml'
+    )
+    await wait(8000)
+
+    // write new yml containing github url
+    await writeFile('./cli-test.yml', changeFile)
```

Comment:
Maybe `cli-test.yml` needs to be added to .gitignore so that future dev won't accidentally commit it to the repo.

Reply:
No need, because this library creates completely enclosed environment for each test, including the filesystem. So no real file really created

![image](https://user-images.githubusercontent.com/5974862/180115549-78dc4c63-b79b-49e7-b544-c444b72694fd.png)

Reply:
oh ok cool!